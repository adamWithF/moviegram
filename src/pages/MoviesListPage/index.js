import React, { PureComponent } from "react"
import propTypes from "prop-types"
import { connect } from "react-redux"

import List from "../../components/List"
import Layout from "../../components/Layout"
import MovieRecord from "../../components/MovieRecord"
import Loader from "../../components/Loader"

import { movieActions } from "../../actions/actions"
import { moviesSelector, movieLoadingSelector } from "../../selectors"

export class MoviesListPage extends PureComponent {
  static propTypes = {
    isLoading: propTypes.bool.isRequired,
    movies: propTypes.array.isRequired,
    getMovies: propTypes.func.isRequired
  }

  state = {
    searchResult: null
  }

  componentDidMount() {
    this.props.getMovies()
  }

  searchHandler = searchResult => this.setState({ searchResult })

  render() {
    const { isLoading, movies } = this.props
    const { searchResult } = this.state
    const dataSource = searchResult ? searchResult : movies

    return (
      <Layout searchHandler={this.searchHandler} backActive={false}>
        <div className="movies-list-page page">
          {isLoading ? (
            <Loader />
          ) : (
            <List
              className="movies-list"
              list={dataSource}
              Item={MovieRecord}
            />
          )}
        </div>
      </Layout>
    )
  }
}

const mapStateToProps = state => ({
  movies: moviesSelector(state),
  isLoading: movieLoadingSelector(state)
})

const mapDispatchToProps = dispatch => ({
  getMovies: () => dispatch(movieActions.getMovies())
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MoviesListPage)
