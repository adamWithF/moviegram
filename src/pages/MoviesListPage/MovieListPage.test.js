import React from "react"
import { shallow } from "enzyme"

import { MoviesListPage } from "../MoviesListPage"

const movieStab = {
  image_url: "http://image.com",
  title: "movie title",
  showtimes: [new Date(), new Date()],
  description: "movie description"
}

describe("MoviesListPage", () => {
  let props

  beforeEach(() => {
    props = {
      movies: [],
      isLoading: false,
      getMovies: () => null
    }
  })

  it("should render properly when loading", () => {
    props.isLoading = true
    const wrapper = shallow(<MoviesListPage {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it("should render properly when no movies", () => {
    const wrapper = shallow(<MoviesListPage {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it("should render properly when movies exists", () => {
    props.movies = [movieStab, movieStab]
    const wrapper = shallow(<MoviesListPage {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
