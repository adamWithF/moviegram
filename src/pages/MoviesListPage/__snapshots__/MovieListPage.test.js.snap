// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesListPage should render properly when loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MoviesListPage
    getMovies={[Function]}
    isLoading={true}
    movies={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backActive": false,
      "children": <div
        className="movies-list-page page"
>
        <Loader />
</div>,
      "searchHandler": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loader />,
        "className": "movies-list-page page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backActive": false,
        "children": <div
          className="movies-list-page page"
>
          <Loader />
</div>,
        "searchHandler": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Loader />,
          "className": "movies-list-page page",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`MoviesListPage should render properly when movies exists 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MoviesListPage
    getMovies={[Function]}
    isLoading={false}
    movies={
        Array [
            Object {
              "description": "movie description",
              "image_url": "http://image.com",
              "showtimes": Array [
                2019-05-23T10:17:39.133Z,
                2019-05-23T10:17:39.133Z,
              ],
              "title": "movie title",
            },
            Object {
              "description": "movie description",
              "image_url": "http://image.com",
              "showtimes": Array [
                2019-05-23T10:17:39.133Z,
                2019-05-23T10:17:39.133Z,
              ],
              "title": "movie title",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backActive": false,
      "children": <div
        className="movies-list-page page"
>
        <List
                Item={[Function]}
                className="movies-list"
                itemName="record"
                list={
                        Array [
                                Object {
                                  "description": "movie description",
                                  "image_url": "http://image.com",
                                  "showtimes": Array [
                                    2019-05-23T10:17:39.133Z,
                                    2019-05-23T10:17:39.133Z,
                                  ],
                                  "title": "movie title",
                                },
                                Object {
                                  "description": "movie description",
                                  "image_url": "http://image.com",
                                  "showtimes": Array [
                                    2019-05-23T10:17:39.133Z,
                                    2019-05-23T10:17:39.133Z,
                                  ],
                                  "title": "movie title",
                                },
                              ]
                }
        />
</div>,
      "searchHandler": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <List
          Item={[Function]}
          className="movies-list"
          itemName="record"
          list={
                    Array [
                              Object {
                                "description": "movie description",
                                "image_url": "http://image.com",
                                "showtimes": Array [
                                  2019-05-23T10:17:39.133Z,
                                  2019-05-23T10:17:39.133Z,
                                ],
                                "title": "movie title",
                              },
                              Object {
                                "description": "movie description",
                                "image_url": "http://image.com",
                                "showtimes": Array [
                                  2019-05-23T10:17:39.133Z,
                                  2019-05-23T10:17:39.133Z,
                                ],
                                "title": "movie title",
                              },
                            ]
          }
/>,
        "className": "movies-list-page page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "Item": [Function],
          "className": "movies-list",
          "itemName": "record",
          "list": Array [
            Object {
              "description": "movie description",
              "image_url": "http://image.com",
              "showtimes": Array [
                2019-05-23T10:17:39.133Z,
                2019-05-23T10:17:39.133Z,
              ],
              "title": "movie title",
            },
            Object {
              "description": "movie description",
              "image_url": "http://image.com",
              "showtimes": Array [
                2019-05-23T10:17:39.133Z,
                2019-05-23T10:17:39.133Z,
              ],
              "title": "movie title",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backActive": false,
        "children": <div
          className="movies-list-page page"
>
          <List
                    Item={[Function]}
                    className="movies-list"
                    itemName="record"
                    list={
                              Array [
                                        Object {
                                          "description": "movie description",
                                          "image_url": "http://image.com",
                                          "showtimes": Array [
                                            2019-05-23T10:17:39.133Z,
                                            2019-05-23T10:17:39.133Z,
                                          ],
                                          "title": "movie title",
                                        },
                                        Object {
                                          "description": "movie description",
                                          "image_url": "http://image.com",
                                          "showtimes": Array [
                                            2019-05-23T10:17:39.133Z,
                                            2019-05-23T10:17:39.133Z,
                                          ],
                                          "title": "movie title",
                                        },
                                      ]
                    }
          />
</div>,
        "searchHandler": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <List
            Item={[Function]}
            className="movies-list"
            itemName="record"
            list={
                        Array [
                                    Object {
                                      "description": "movie description",
                                      "image_url": "http://image.com",
                                      "showtimes": Array [
                                        2019-05-23T10:17:39.133Z,
                                        2019-05-23T10:17:39.133Z,
                                      ],
                                      "title": "movie title",
                                    },
                                    Object {
                                      "description": "movie description",
                                      "image_url": "http://image.com",
                                      "showtimes": Array [
                                        2019-05-23T10:17:39.133Z,
                                        2019-05-23T10:17:39.133Z,
                                      ],
                                      "title": "movie title",
                                    },
                                  ]
            }
/>,
          "className": "movies-list-page page",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "Item": [Function],
            "className": "movies-list",
            "itemName": "record",
            "list": Array [
              Object {
                "description": "movie description",
                "image_url": "http://image.com",
                "showtimes": Array [
                  2019-05-23T10:17:39.133Z,
                  2019-05-23T10:17:39.133Z,
                ],
                "title": "movie title",
              },
              Object {
                "description": "movie description",
                "image_url": "http://image.com",
                "showtimes": Array [
                  2019-05-23T10:17:39.133Z,
                  2019-05-23T10:17:39.133Z,
                ],
                "title": "movie title",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`MoviesListPage should render properly when no movies 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MoviesListPage
    getMovies={[Function]}
    isLoading={false}
    movies={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backActive": false,
      "children": <div
        className="movies-list-page page"
>
        <List
                Item={[Function]}
                className="movies-list"
                itemName="record"
                list={Array []}
        />
</div>,
      "searchHandler": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <List
          Item={[Function]}
          className="movies-list"
          itemName="record"
          list={Array []}
/>,
        "className": "movies-list-page page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "Item": [Function],
          "className": "movies-list",
          "itemName": "record",
          "list": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backActive": false,
        "children": <div
          className="movies-list-page page"
>
          <List
                    Item={[Function]}
                    className="movies-list"
                    itemName="record"
                    list={Array []}
          />
</div>,
        "searchHandler": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <List
            Item={[Function]}
            className="movies-list"
            itemName="record"
            list={Array []}
/>,
          "className": "movies-list-page page",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "Item": [Function],
            "className": "movies-list",
            "itemName": "record",
            "list": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
